samples/drivers/clock_control_xec/src/main.c

```diff
@@ -179,8 +181,10 @@ static void vbat_power_fail(void)
 }
 #endif
 
-static const struct gpio_ctrl_regs * const gpc =
-	(struct gpio_ctrl_regs *)(DT_REG_ADDR(DT_NODELABEL(gpio_000_036)));
+#define MCHP_XEC_PINCFG(pin) ((pin >> 5) * 32 + (pin & 0x1f))
```

Comment:
can we move this macro to the soc layer?

Reply:
done.